### PR TITLE
[FIX] l10n_cl: prevent IndexError when creating customer

### PR DESCRIPTION
**Pull request by reka-odoo**
When the user creates a new customer and enters "." in Identification Number,
a traceback will appear.

Steps to reproduce the error:
- Install "l10n_cl" module
- Switch to Chile Company
- Go to Invoicing > Customers > Create New Customer > Enter Name >
  Select 'Chile' in Country > Enter "." in Identification Number > Save

Traceback:
```
IndexError: string index out of range
  File "odoo/http.py", line 2139, in __call__
    response = request._serve_db()
  File "odoo/http.py", line 1715, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "odoo/service/model.py", line 133, in retrying
    result = func()
  File "odoo/http.py", line 1742, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "odoo/http.py", line 1943, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "odoo/addons/base/models/ir_http.py", line 191, in _dispatch
    result = endpoint(**request.params)
  File "odoo/http.py", line 717, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "addons/web/controllers/dataset.py", line 30, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "addons/web/controllers/dataset.py", line 26, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "odoo/api.py", line 464, in call_kw
    result = _call_kw_model_create(method, model, args, kwargs)
  File "odoo/api.py", line 444, in _call_kw_model_create
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-490>", line 2, in create
  File "odoo/api.py", line 414, in _model_create_multi
    return create(self, [arg])
  File "addons/l10n_cl/models/res_partner.py", line 55, in create
    vals['vat'] = self._format_vat_cl(vals)
  File "addons/l10n_cl/models/res_partner.py", line 41, in _format_vat_cl
    return stdnum.util.get_cc_module('cl', 'vat').format(values['vat']).replace('.', '').replace(
  File "stdnum/cl/rut.py", line 90, in format
    number[-4:-1] + '-' + number[-1])
```

code reference:
https://github.com/odoo/odoo/blob/276972f96d2eb1a65686e7319f601ed435b57b88/addons/l10n_cl/models/res_partner.py#L40
Here, when user enters "." in Identification Number
then values['vat'] will be ".",
So when it will try to .format(),
It will lead to above traceback.

sentry-4593791704

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by reka-odoo** [comment]
Hello @odoo/sentry_reviewers 
Please review this PR.
Thanks!

File: addons/l10n_cl/models/res_partner.py [hunk 1]
```diff
@@ -37,8 +37,11 @@ def _format_vat_cl(self, values):
         partner_country_is_chile = country.code == "CL" or identification_type.country_id.code == "CL"
         if partner_country_is_chile and \
                 values.get('l10n_latam_identification_type_id') in identification_types and values.get('vat'):
-            return stdnum.util.get_cc_module('cl', 'vat').format(values['vat']).replace('.', '').replace(
-                'CL', '').upper()
+            try:
+                return stdnum.util.get_cc_module('cl', 'vat').format(values['vat']).replace('.', '').replace(
+                    'CL', '').upper()
+            except IndexError:
+                return values['vat']
```
Comment (antoine162) [inline addons/l10n_cl/models/res_partner.py:44]: Sorry for the long wait! How about doing this?
```suggestion
        if partner_country_is_chile and \
                values.get('l10n_latam_identification_type_id') in identification_types and values.get('vat') and\
                stdnum.util.get_cc_module('cl', 'vat').is_valid(values['vat']):
            return stdnum.util.get_cc_module('cl', 'vat').format(values['vat']).replace('.', '').replace(
                    'CL', '').upper()
```
see the is_valid function in python-stdnum -> https://github.com/arthurdejong/python-stdnum/blob/master/stdnum/cl/rut.py

Comment (reka-odoo) [reply addons/l10n_cl/models/res_partner.py]: Hello @antoine162 
Changes are done!
It is working properly.
Please review it.
Thanks!

**Review by antoine162 — APPROVED** [review]
LGTM :+1: 
@jco-odoo

**Comment by jco-odoo** [comment]
@Danisan fyi

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by reka-odoo** [comment]
@robodoo retry

**Pull request closed without merging**